luarules/gadgets/cmd_factory_stop_production.lua

```diff
@@ -46,41 +47,62 @@ end
 --------------------------------------------------------------------------------
 
 local stopProductionCmdDesc = {
-	id      = CMD_STOP_PRODUCTION,
-	type    = CMDTYPE.ICON,
-	name    = 'Stop Production',
-	action  = 'stopproduction',
-	cursor  = 'Stop', -- Probably does nothing
-	tooltip = 'Stop Production: Clear factory production queue.',
+	id = CMD_STOP_PRODUCTION,
+	type = CMDTYPE.ICON,
+	name = "Stop Production",
+	action = "stopproduction",
+	cursor = "Stop", -- Probably does nothing
+	tooltip = "Stop Production: Clear factory production queue.",
+	hidden = false,
+	queueing = false,
 }
 
 --------------------------------------------------------------------------------
 --------------------------------------------------------------------------------
 -- Handle the command
 
 function gadget:AllowCommand_GetWantedCommand()
-	return {[CMD_STOP_PRODUCTION] = true}
+	return { [CMD_STOP_PRODUCTION] = true }
 end
 
 function gadget:AllowCommand_GetWantedUnitDefID()
 	return isFactory
 end
 
-function gadget:AllowCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOptions)
-	if (cmdID ~= CMD_STOP_PRODUCTION) or (not isFactory[unitDefID]) then
+function gadget:AllowCommand(unitID, unitDefID, _, cmdID)
```

Comment:
Add the real name of the variable, I understand the meaning of the `_` but this way it is complicated for other people to provide maintenance.

Reply:
Does it help if I put the full signature commented above?

Reply:
e.g.

```lua
// function gadget:AllowCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOptions)
function gadget:AllowCommand(unitID, unitDefID, _, cmdID)
```